noether_simulator/src/noether_simulator.cpp

```diff
@@ -0,0 +1,393 @@
+/*
+ * Copyright (c) 2016, Southwest Research Institute
+ * All rights reserved.
+ *
+ */
+
+#include <noether_simulator/noether_simulator.h>
+
+#include <Eigen/Geometry>
+#include <vtkCylinderSource.h>
+#include <vtkDoubleArray.h>
+
+#include <vtkTransform.h>
+#include <vtkTransformPolyDataFilter.h>
+
+#include <vtkDelaunay3D.h>
+#include <vtkUnstructuredGrid.h>
+#include <vtkGeometryFilter.h>
+
+#include <vtkExtractPoints.h>
+#include <vtkSelectEnclosedPoints.h>
+#include <vtkModifiedBSPTree.h>
+#include <vtkLookupTable.h>
+
+namespace noether_simulator
+{
+  NoetherSimulator::NoetherSimulator()
+  {
+    process_base_rate_ = 5.0;
+    scalar_sigma_ = 3.0;
+    debug_on_ = false;
+    simulation_points_ = vtkSmartPointer<vtkPolyData>::New();
+  }
+
+  double NoetherSimulator::integral(double pt[3])
+  {
+    //double k = process_base_rate_;
+    double result;
+    // t is not currently added/used, but can be so as to allow time parameterization between process points
+    // f(x,y,z,t) = -x^2 - y^2 - (z - k)
+    // integral(f) = -x^3*y*z/3 - x*y^3*z/3 - x*y*z^2/2 + k*x*y*z
+    // TODO: the above equation does not work (integrates area, not along a line)
+    // TODO: again, cylinder is along y-axis, so y and z are flipped
+      result = -( pow(pt[0],3.0)) /3.0 - (pow(pt[2],3.0))/3.0 ;
+
+    return 10*result;
+  }
+
+  double NoetherSimulator::calculateIntegration(vtkSmartPointer<vtkPoints> points)
+  {
+    if(points->GetNumberOfPoints() < 2)
+      return 0;
+    // This is a place holder equation for the tool and is process specific
+    // TODO: create a more generic interface to allow for other tools to be created and used
+
+    double pt1[3], pt2[3];
+    double val1, val2;
+
+    points->GetPoint(0, pt1);
+    points->GetPoint(1, pt2);
+
+    val1 = integral(pt1);
+    val2 = integral(pt2);
+
+    double value = (val2 - val1);
+    return fabs(value);
+  }
+
+vtkSmartPointer<vtkPolyData>  NoetherSimulator::getSimulatedPoints()
+  {
+    vtkSmartPointer<vtkPolyData> simulated_points = simulation_points_;
+    return simulated_points;
+  }
+
+  void  NoetherSimulator::runSimulation()
+  {
+
+    cout << "starting simulation\n";
```

Comment:
Use one of the logging functions instead of cout directly so that output can be turned on and off level by level. (e.g. ROS_ERROR, ROS_INFO, and ROS_DEBUG)